src/schema/v1.json

```diff
@@ -3,63 +3,121 @@
  "type":"object",
- "title":"CoreOS v1 meta schema",
- "required": [
-   "buildid",
-   "coreos-assembler.basearch",
-   "coreos-assembler.build-timestamp",
-   "coreos-assembler.code-source",
-   "coreos-assembler.config-dirty",
-   "coreos-assembler.config-gitrev",
-   "coreos-assembler.image-config-checksum",
-   "coreos-assembler.image-genver",
-   "coreos-assembler.image-input-checksum",
-   "images",
-   "name",
-   "ostree-commit",
-   "ostree-content-bytes-written",
-   "ostree-content-checksum",
-   "ostree-n-cache-hits",
-   "ostree-n-content-total",
-   "ostree-n-content-written",
-   "ostree-n-metadata-total",
-   "ostree-n-metadata-written",
-   "ostree-timestamp",
-   "ostree-version",
-   "rpm-ostree-inputhash",
-   "summary"
-  ],
- "optional": [
-   "aliyun",
-   "amis",
-   "azure",
-   "build-url",
-   "gcp",
-   "oscontainer",
-   "pkgdiff",
-   "coreos-assembler.container-config-git",
-   "coreos-assembler.container-image-git"
-  ],
+ "title":"CoreOS Assember v1 meta.json schema",
+ "allOf": [
+     {
+         "required": [
+             "buildid",
+             "name",
+             "ostree-commit",
+             "ostree-content-bytes-written",
+             "ostree-content-checksum",
+             "ostree-n-cache-hits",
+             "ostree-n-content-total",
+             "ostree-n-content-written",
+             "ostree-n-metadata-total",
+             "ostree-n-metadata-written",
+             "ostree-timestamp",
+             "ostree-version",
+             "rpm-ostree-inputhash",
+             "summary"
+         ],
+         "optional": [
+           "images",
+           "aliyun",
+           "amis",
+           "azure",
+           "build-url",
+           "gcp",
+           "oscontainer",
+           "pkgdiff",
+
+           "coreos-assembler.basearch",
+           "coreos-assembler.build-timestamp",
+           "coreos-assembler.code-source",
+           "coreos-assembler.config-dirty",
+           "coreos-assembler.config-gitrev",
+           "coreos-assembler.container-config-git",
+           "coreos-assembler.container-image-git",
+           "coreos-assembler.image-config-checksum",
+           "coreos-assembler.image-genver",
+           "coreos-assembler.image-input-checksum",
+           "fedora-coreos.parent-commit",
+           "fedora-coreos.parent-version",
+           "ref"
+          ]
+     },
+     {
+         "if": {
+             "properties": {
+                 "name": { "const": "fedora-coreos"  }
+             }
+         },
+         "then": {
+                "required": [
+                   "ref"
+                  ],
+                "optional": [
+                  "fedora-coreos.parent-commit",
+                  "fedora-coreos.parent-version"
+                ]
+         },
+         "else": {}
+    },
+    {
+         "if": {
+             "properties": {
+                 "name": { "pattern": "^(rhcos|fedora-coreos)"  }
```

Comment:
Many of these are core properties emitted by coreos-assembler; I don't see why e.g. if someone is doing a custom build they shouldn't have `coreos-assembler.basearch`.